virtio-devices/src/iommu.rs

```diff
@@ -554,37 +554,41 @@ impl Request {
                         .collect();
 
                     // Trigger external unmapping if necessary.
+                    // We don't know which endpoint this mapping is for so
+                    // we attempt to unmap on all endpoints and ignore errors
                     for endpoint in endpoints {
                         if let Some(ext_map) = ext_mapping.get(&endpoint) {
                             let size = req.virt_end - virt_start + 1;
-                            ext_map
-                                .unmap(virt_start, size)
-                                .map_err(Error::ExternalUnmapping)?;
+                            let _ = ext_map.unmap(virt_start, size);
```

Comment:
It seems unsafe to ignore errors? The guest application may assume that the devices are isolated even if they are not.

Reply:
So I debated about this one for a while because I agree I don't like ignoring these errors - it's a little difficult as the endpoints aren't aren't included as part of the map requests. As a result, you have to iterate over each endpoint until you get a successful pass. However what I can do is on the original map request, detect which endpoint the mapping was successfully executed on, then update the domain mapping btree to include the endpoint associated with each mapping - does that sound good?